model-optimizer/extensions/front/tf/InterpolateTransposes.py

```diff
@@ -16,49 +16,56 @@
 
-    pattern_nodes = [
-        ('interpolate', {'kind': 'op', 'op': 'Interpolate'}),
-        ('transpose_1', {'kind': 'op', 'op': 'Transpose'}),
-        ('transpose_1_order', {'kind': 'op', 'op': 'Const',
-                               'value': lambda x: x is not None and np.array_equal(x, int64_array([0, 2, 3, 1]))}),
-        ('transpose_2', {'kind': 'op', 'op': 'Transpose'}),
-        ('transpose_2_order', {'kind': 'op', 'op': 'Const',
-                               'value': lambda x: x is not None and np.array_equal(x, int64_array([0, 3, 1, 2]))}),
-    ]
-    pattern_edges = [
-        ('transpose_1', 'interpolate', {'in': 0, 'out': 0}),
-        ('transpose_1_order', 'transpose_1', {'in': 1, 'out': 0}),
-        ('interpolate', 'transpose_2', {'in': 0, 'out': 0}),
-        ('transpose_2_order', 'transpose_2', {'in': 1, 'out': 0}),
-    ]
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('interpolate', {'kind': 'op', 'op': 'Interpolate'}),
+                ('transpose_1', {'kind': 'op', 'op': 'Transpose'}),
+                ('transpose_1_order', {'kind': 'op', 'op': 'Const'}),
+                ('transpose_2', {'kind': 'op', 'op': 'Transpose'}),
+                ('transpose_2_order', {'kind': 'op', 'op': 'Const'}),
+            ],
+            edges=[
+                ('transpose_1', 'interpolate', {'in': 0, 'out': 0}),
+                ('transpose_1_order', 'transpose_1', {'in': 1, 'out': 0}),
+                ('interpolate', 'transpose_2', {'in': 0, 'out': 0}),
+                ('transpose_2_order', 'transpose_2', {'in': 1, 'out': 0}),
+            ]
+        )
 
-    def transform_graph(self, graph: Graph, replacement_descriptions: dict):
-        matches = find_pattern_matches(graph, self.pattern_nodes, self.pattern_edges)
-        for match in list(matches):
-            inverse_match = inverse_dict(match)
-            interpolate = Node(graph, inverse_match['interpolate'])
-            transpose_1 = Node(graph, inverse_match['transpose_1'])
-            transpose_2 = Node(graph, inverse_match['transpose_2'])
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        interpolate = match['interpolate']
+        transpose_1 = match['transpose_1']
+        transpose_2 = match['transpose_2']
+        transpose_1_order = match['transpose_1_order']
+        transpose_2_order = match['transpose_2_order']
 
-            # because we remove Transpose layers the ResizeNearestNeighbor should be updated for NCHW layout
-            interpolate.axes = int64_array([2, 3])
+        if not (transpose_1_order.has_valid('value') and transpose_2_order.has_valid('value')):
+            return
 
-            transpose_1.in_port(0).get_connection().set_destination(interpolate.in_port(0))
-            transpose_2.out_port(0).get_connection().set_source(interpolate.out_port(0))
+        if not np.array_equal(transpose_1_order.value, int64_array([0, 2, 3, 1])):
```

Comment:
Why did you moved this out of pattern?

Reply:
Because this transformation was not called.

Reply:
We need to understand why this transformation was not called and make it work. I don't like the idea to have a checks inside the replacement function which can be a part of the pattern.